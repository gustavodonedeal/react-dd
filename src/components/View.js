import React from 'react';
import axios from 'axios';
import PriceTag from './PriceTag';
import MediaPreview from './MediaPreview';
import SellerDetails from './SellerDetails';

const corsUrlPrefix = 'https://cors-anywhere.herokuapp.com/';
const viewApiUrl = `${corsUrlPrefix}https://www.donedeal.ie/cadview/api/v3/view/ad/`;

class View extends React.Component {

  constructor() {
    super();

    this.state = {
      ad: {}
    }
  };

  componentDidMount() {
    this.serverRequest = axios.get(viewApiUrl + this.props.match.params.adId)
      .then((response) => {
        this.setState({
          ad: response.data,
        });
      }).catch((error) => {
        console.error(error);
      });
  };

  render() {
    const ad = this.state.ad;

    return (
      <div className="App">
        <h1>Ad Details</h1>

        <h2>{ad.header}</h2>
        <description className="desc">
          {ad.description}
        </description>
        <div>
          <PriceTag ad={ad} />
          {ad.county}
        </div>

        <MediaPreview ad={ad} />

        <SellerDetails ad={ad} />

      </div>
    )
  }
}

export default View;
