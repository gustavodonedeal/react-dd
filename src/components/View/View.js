import React from 'react';
import PriceTag from '../PriceTag';
import MediaPreview from '../MediaPreview';
import SellerDetails from '../SellerDetails';
import { getAd } from '../../api/ads';

class View extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      ad: {}
    }
  };

  componentDidMount() {
    const { adId } = this.props.match.params;
    getAd(adId).then(ad => 
      this.setState(prevState => ({ ad }))
    );
  };

  render() {
    const { ad } = this.state;
    return (
      <div className="App">
        <h1>Ad Details</h1>
        <h2>{ad.header}</h2>
        <description className="desc">
          {ad.description}
        </description>
        <div>
          <PriceTag ad={ad} />
          {ad.county}
        </div>
        <MediaPreview ad={ad} />
        <SellerDetails ad={ad} />
      </div>
    )
  }
}

export default View;
