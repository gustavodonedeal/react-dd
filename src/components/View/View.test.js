import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import fetchMock from 'fetch-mock';
import View from './View';

it('renders "empty ad" correctly', async () => {
  const json = await import('../../../__tests__/sampleAdvert.json');
  fetchMock.mock('*', [json]);
  const context = {};
  const tree = renderer.create(
    <View match={{ params: { adId: 16140864 }}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
  fetchMock.restore();
});