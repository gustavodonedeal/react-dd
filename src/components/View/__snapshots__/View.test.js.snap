// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "empty ad" correctly 1`] = `
<div
  className="App"
>
  <h1>
    Ad Details
  </h1>
  <h2 />
  <description
    className="desc"
  />
  <div>
    <p
      className="card__price"
    >
      <span>
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "No Price",
            }
          }
        />
        
      </span>
    </p>
  </div>
  <div
    className="card__media"
  >
    <div
      className="card__photo"
    >
      <img
        alt={undefined}
        src={null}
      />
    </div>
    <div
      className="card__media-count"
    >
      <i
        className="icon-camera"
      />
      1/
    </div>
  </div>
  <div
    className="seller-interaction-container"
  >
    <div
      className="contact-seller-container"
    >
      <div
        className="contact-seller-header"
      >
        <span
          className="contact-seller-image icon-user_tick"
        />
        <div
          className="contact-seller-title"
        >
          <span
            className="contact-seller-name"
          >
            
          </span>
        </div>
      </div>
      <div
        className="contact-seller-method contact-seller-by-message disabled"
      >
        <div
          className="contact-seller-method-content"
        >
          <button
            className="btn contact-seller-message-submit-btn"
            disabled="disabled"
            id="contact-seller-send-btn-main"
          >
            Seller can't be messaged
          </button>
        </div>
      </div>
      <div
        className="contact-seller-method contact-seller-by-phone"
      >
        <div
          className="btn contact-seller-by-phone-btn input-btn"
        >
          <button
            className="show-number"
            id="show-number-btn"
            onClick={[Function]}
            title="Click to show phone number"
          >
            Seller can't be phoned
          </button>
        </div>
      </div>
    </div>
    <div
      className="seller-details-container"
    >
      <div
        className="seller-details-content"
      >
        <dl
          className="seller-detail"
        >
          <dt
            className="seller-detail-label"
          >
            Location
          </dt>
          <dd
            className="seller-detail-value"
          >
            <address>
              ---
            </address>
          </dd>
        </dl>
        <dl
          className="seller-detail"
        >
          <dt
            className="seller-detail-label"
          >
            DoneDealing since
          </dt>
          <dd
            className="seller-detail-value"
          >
            2018
          </dd>
        </dl>
        <dl
          className="seller-detail"
        >
          <dt
            className="seller-detail-label"
          >
            Listed ads
          </dt>
          <dd
            className="seller-detail-value"
          >
            
          </dd>
        </dl>
        <div
          className="seller-verification-status"
        >
          <span
            className="verification-credential email"
          >
            <span
              className="verified"
            >
              <i
                className="icon icon-tick_green"
              />
              Email Verified
            </span>
          </span>
          <span
            className="verification-credential phone"
          >
            <span
              className="verified"
            >
              <i
                className="icon icon-tick_green"
              />
              Phone Verified
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
